js/map_utils.js

```diff
@@ -36,15 +44,21 @@ function addStates(map, stateBounds) {
 }
 
 // on click
-function zoomToFromState(data) {
-
+function zoomToFromState(d, selection = null) {
+  
+  if(typeof selection === "number") { // because `i` is defaulted
+    // couldn't rely on this when I changed from `.on("click", zoomToFromState)`
+    // to `.on("click", function(d) { zoomToFromState(); })`
+    selection = d3.select(this);
+  } 
```

Comment:
so what happens if selection is null? then we proceed to call things like `selection.attr()`? seems like that would lead to errors, wouldn't it?

Reply:
Well, it shouldn't ever be null. I need to remove the default, I think. If you use this function like `.on("click", zoomToFromState)`, d is automatically `d` and selection is automatically `i` as in `function(d,i)`. So selection should never be null. Otherwise, it is being use where you explicitly are passing d and selection in, see https://github.com/USGS-VIZLAB/water-use-15/pull/184/files/300afba47e390b96bf1c6fbd5857de8f8abed8c6#diff-dbac7df67f73b73347fbe15446a0ba2dR133